### ISSUE TITLE
False positive #[Override] required for __toString()

### ISSUE THREAD

**Issue by zerkms** [vimeo/psalm#10846]
It's not reproducible at psalm-dev, but here is a repro repository: https://github.com/zerkms/psalm-10846-repro

```php
<?php
declare(strict_types=1);

namespace Zerkms;

final readonly class Test
{
    public function __toString(): string
    {
        return '';
    }
}
```
```
ERROR: MissingOverrideAttribute - src/Test.php:8:5 - Method __toString should have the "Override" attribute
```

**Comment by edsrzf** [comment]
I think you could argue this either way. Adding the `#[Override]` attribute is valid in that it doesn't cause a PHP runtime error.

I think the behavior I'd shoot for is:

* If `Test` _explicitly_ implements `Stringable`, then `MissingOverrideAttribute` is triggered.
* If `Test` _implicitly_ implements `Stringable`, as in the reproducer, then no `MissingOverrideAttribute` is triggered.

Sound reasonable?

**Comment by zerkms** [comment]
I don't think `Stringable` deserves any special treatment at all, unless the language requires. All exceptions is a burden to maintain by the core developers team.

My suggestion: **ONLY** mark a method as `MissingOverrideAttribute` **IF AND ONLY IF** there is and **EXPLICITLY&& extended type that declares the method with the same name.

**Comment by zerkms** [comment]
PS: kia ora 🥝 :-D

**Comment by weirdan** [comment]
> It's not reproducible at psalm-dev,

You just need to set one of the checkboxes on the settings screen: https://psalm.dev/r/07b8ed2a92
